src/shell/plugins/fence.c

```diff
@@ -102,37 +102,24 @@ static void exchange_exit_cb (struct exchange *xcg, void *arg)
  */
 static int parse_fence_attr (struct fence_call *fxcall, pmix_info_t *info)
 {
-    int rc = PMIX_SUCCESS;
-    int required = (info->flags & PMIX_INFO_REQD);
-
     if (!strcmp (info->key, "pmix.collect")) {
-        if (info->value.type != PMIX_BOOL)
-            goto type_error;
+        if (info->value.type != PMIX_BOOL) {
+            shell_warn ("fence attr %s has wrong type=%d",
+                        info->key,
+                        info->value.type);
+            return PMIX_ERR_BAD_PARAM;
+        }
         if (info->value.data.flag == true)
             fxcall->collect = true;
-        goto done;
     }
-    /* This attr is listed in the openpmix EXCEPTIONS as an
-     * experimental development feature and pops pops up when running
-     * osu benchmarks.  We don't know what to do with it, but let's
-     * tone down the warning to debug level to avoid alarming users.
-     */
-    else if (!strcmp (info->key, "pmix.loc.col.st")) {
-        shell_debug ("ignoring experimental %s fence attr: %s",
-                     required ? "required" : "optional",
-                     info->key);
-        goto done;
+    else if ((info->flags & PMIX_INFO_REQD)) {
+        shell_warn ("unknown required fence attr: %s", info->key);
```

Comment:
There's two error messages that are added here at `shell_warn()`, but the commit message says this commit should be replacing `shell_warn()` calls with `shell_debug()`. Not sure if the commit message just needs an update or if these changes were meant to be in a separate commit? (Or perhaps I'm misreading the commit message)

Reply:
The commit does do what it says, but it's not so clear from the diff because the change allowed me to simplify the function.  There are now three legs to the if clause:  1) a known attribute (includes an error logged at warn level if it's the wrong type), 2) attributes that are required but unknown (logged at warn level), and 3) attributes that are not required and unknown (logged at debug level).

Does that make sense?

Reply:
Works for me!